### PR TITLE
handle mixed state for stretch params (knots)

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request extends on #2606 to support the interactive splines introduced in #2545.  Its a separate PR as #2606 was milestones for a bugfix release that does not include #2545.

https://github.com/spacetelescope/jdaviz/assets/877591/828ee4e3-a9ac-43da-ae28-c45319b55142

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/default/plugins/plot_options/plot_options.py [hunk 1]
```diff
@@ -634,6 +634,12 @@ def vue_unmix_state(self, names):
         for name in names:
             sync_state = getattr(self, name)
             sync_state.unmix_state()
+        if 'stretch_params' in names:
+            # there is no way to call send_state to force the update to the layers,
+            # so we'll force an update by clearing first
+            stretch_params = dict(self.stretch_params_value)
+            self.stretch_params_value = {}
+            self.stretch_params_value = stretch_params
```
Comment (kecnry) [inline jdaviz/configs/default/plugins/plot_options/plot_options.py:642]: I really don't like this, but couldn't find a way around it in this case 🤔

File: jdaviz/core/template_mixin.py [hunk 2]
```diff
@@ -3315,6 +3327,11 @@ def _update_mixed_state(self):
             for state in self.linked_states:
                 current_glue_values.append(self._get_glue_value(state))
             mixed = self.is_mixed(current_glue_values)
+            # ensure the value corresponds to the first entry, this prevents the case where a glue
+            # change to one of the linked_states changes the value that will be adopted when
+            # unmixing something in mixed state and results in more consistent and predictable
+            # behavior
+            self.value = current_glue_values[0]
```
Comment (kecnry) [inline jdaviz/core/template_mixin.py:3334]: I suspect this change is causing the test failures - will need to think how to do this in a way that guarantees the correct type and avoids recursion.

**Review by gibsongreen — APPROVED** [review]
Everything looks good and runs as demoed!

**Review by haticekaratay — APPROVED** [review]
Conducted some manual testing, and everything worked as expected, with no issues encountered. Thanks for working on this.

**Pull request merged**